std-bits/table/src/main/java/org/enso/table/excel/xssfreader/XSSFReaderWorkbook.java

```diff
@@ -0,0 +1,299 @@
+            }
+          } catch (SAXException
+              | IOException
+              | InvalidFormatException
+              | XPathExpressionException e) {
+            throw new RuntimeException(e);
+          }
+        });
+  }
+
+  private synchronized void readShared() {
+    if (readShared) {
+      return;
+    }
+
+    try {
+      withReader(
+          rdr -> {
+            try {
+              rdr.setUseReadOnlySharedStringsTable(true);
+              sharedStrings = rdr.getSharedStringsTable();
+              if (sharedStrings == null) {
+                sharedStrings =
+                    new SharedStrings() {
+                      @Override
+                      public RichTextString getItemAt(int idx) {
+                        return null;
+                      }
+
+                      @Override
+                      public int getCount() {
+                        return 0;
+                      }
+
+                      @Override
+                      public int getUniqueCount() {
+                        return 0;
+                      }
+                    };
+              }
+
+              // Read the styles table and attach the format data
+              var stylesTable = rdr.getStylesTable();
+              styles = new XSSFReaderFormats(stylesTable);
+
+              readShared = true;
+            } catch (InvalidFormatException | IOException e) {
+              throw new RuntimeException(e);
+            }
+          });
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /** Flag that workbook is in 1904 format. */
+  boolean use1904Format() {
+    return use1904DateSystemFlag;
+  }
+
+  private Map<String, SheetInfo> getSheetInfoMap() {
+    return sheetInfoMap;
+  }
+
+  private List<SheetInfo> getSheetInfos() {
+    return sheetInfos;
+  }
+
+  private Map<String, NamedRange> getNamesMap() {
+    return namedRangeMap;
+  }
+
+  @Override
+  public int getNumberOfSheets() {
+    return getSheetInfoMap().size();
+  }
+
+  @Override
+  public int getSheetIndex(String name) {
+    var map = getSheetInfoMap();
+    if (!map.containsKey(name)) {
+      return -1;
+    }
+    return map.get(name).index;
+  }
+
+  @Override
+  public String getSheetName(int sheet) {
+    var list = getSheetInfos();
+    if (sheet < 0 || sheet >= list.size()) {
+      throw new IllegalArgumentException("Sheet index out of range: " + sheet);
+    }
+    return list.get(sheet).name;
+  }
+
+  @Override
+  public int getNumberOfNames() {
+    return getNamesMap().size();
+  }
+
+  @Override
+  public String[] getRangeNames() {
+    return getNamesMap().keySet().toArray(String[]::new);
+  }
+
+  @Override
+  public String getNameFormula(String name) {
+    var map = getNamesMap();
+    var namedRange = map.get(name);
+    return namedRange == null ? null : namedRange.formula;
+  }
+
+  public SharedStrings getSharedStrings() {
+    if (!readShared) {
+      readShared();
+    }
```

Comment:
`readShared()` also does the `if(readShared)` check at the beginning of its body, so these checks are redundant.

Why don't we rename it to `ensureSharedIsRead()` and just do:
```suggestion
    ensureSharedIsRead()
```